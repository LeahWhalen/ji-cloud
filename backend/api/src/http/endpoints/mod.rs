pub mod admin;
pub mod audio;
pub mod category;
pub mod image;
pub mod jig;
pub mod meta;
pub mod module;
pub mod user;
pub mod animation;
