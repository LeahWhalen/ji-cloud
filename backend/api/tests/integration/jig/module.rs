use http::StatusCode;
use serde_json::json;

use crate::{
    fixture::Fixture,
    helpers::{initialize_server, LoginExt},
};

#[actix_rt::test]
async fn update_empty() -> anyhow::Result<()> {
    let app = initialize_server(&[Fixture::User, Fixture::Modules]).await;

    let port = app.port();

    let client = reqwest::Client::new();

    let resp = client
        .patch(&format!(
            "http://0.0.0.0:{}/v1/module/5c8b6458-8840-11eb-8cf7-9bf404dc7386",
            port
        ))
        .login()
        .send()
        .await?
        .error_for_status()?;

    assert_eq!(resp.status(), StatusCode::NO_CONTENT);

    let resp = client
        .get(&format!(
            "http://0.0.0.0:{}/v1/module/5c8b6458-8840-11eb-8cf7-9bf404dc7386",
            port
        ))
        .login()
        .send()
        .await?
        .error_for_status()?;

    assert_eq!(resp.status(), StatusCode::OK);

    let body: serde_json::Value = resp.json().await?;

    insta::assert_json_snapshot!(body, {".**.updated_at" => "[timestamp]"});

    Ok(())
}

#[actix_rt::test]
async fn update_contents() -> anyhow::Result<()> {
    let app = initialize_server(&[Fixture::User, Fixture::Modules]).await;

    let port = app.port();

    let client = reqwest::Client::new();

    let resp = client
        .patch(&format!(
            "http://0.0.0.0:{}/v1/module/5c8b6458-8840-11eb-8cf7-9bf404dc7386",
            port
        ))
        .login()
        .json(&json!({"body": {}}))
        .send()
        .await?
        .error_for_status()?;

    assert_eq!(resp.status(), StatusCode::NO_CONTENT);

    let resp = client
        .get(&format!(
            "http://0.0.0.0:{}/v1/module/5c8b6458-8840-11eb-8cf7-9bf404dc7386",
            port
        ))
        .login()
        .send()
        .await?
        .error_for_status()?;

    assert_eq!(resp.status(), StatusCode::OK);

    let body: serde_json::Value = resp.json().await?;

    insta::assert_json_snapshot!(body, {".**.updated_at" => "[timestamp]"});

    Ok(())
}
