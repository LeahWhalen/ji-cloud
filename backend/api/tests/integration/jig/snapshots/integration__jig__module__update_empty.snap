---
source: tests/integration/jig/module.rs
expression: body

---
{
  "module": {
    "id": "0cbfdd82-7c83-11eb-9f77-d7d86264c3bc",
    "body": {
      "cover": {
        "content": null
      }
    },
    "is_complete": false,
    "is_updated": true,
    "created_at": "2021-03-04T00:46:26.134651Z",
    "updated_at": "[timestamp]"
  }
}
